File: tests/layers/fixtures/repo_python/logger.py
"""Structured logger wrapper for the example app."""
from __future__ import annotations

import json
import sys
from typing import Any


def log_event(level: str, msg: str, **fields: Any) -> None:
    """Emit a single JSON log line."""
    record = {"level": level, "msg": msg, **fields}
    sys.stdout.write(json.dumps(record) + "\n")


def info(msg: str, **fields: Any) -> None:
    log_event("info", msg, **fields)


def error(msg: str, **fields: Any) -> None:
    log_event("error", msg, **fields)

File: tests/layers/fixtures/repo_rust/logger.rs
//! Stdout structured logger.

pub struct Logger {
    pub level: String,
}

impl Logger {
    pub fn new(level: &str) -> Self {
        Logger { level: level.to_string() }
    }

    pub fn info(&self, msg: &str) {
        println!("[{}] {}", self.level, msg);
    }

    pub fn error(&self, msg: &str) {
        eprintln!("[{}] error: {}", self.level, msg);
    }
}
